### PR TITLE
Add util functions to read/write manifest file and update sub commands

### PR DESCRIPTION
**Pull request by Liujingfang1**
This change removes all the dependency on package tree from sub commands add resource, add configmap, set nameprefix

### CODE REVIEW

File: pkg/kinflate/commands/util.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	"strings"
+
+	"github.com/ghodss/yaml"
+
+	manifest "k8s.io/kubectl/pkg/apis/manifest/v1alpha1"
+	"k8s.io/kubectl/pkg/kinflate/constants"
+	"k8s.io/kubectl/pkg/kinflate/util/fs"
+)
+
+func validateManifestFile(mPath string, fsys fs.FileSystem) (string, error) {
```
Comment (mengqiy) [inline pkg/kinflate/commands/util.go]: I'd be happier if you make it Object-orientated by making FileSystem a member of the struct.
@apelisse WDYT?

File: pkg/kinflate/commands/util.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	f, err := fsys.Stat(mPath)
+	if err != nil {
+		errorMsg := fmt.Sprintf("Manifest (%s) missing\nRun `kinflate init` first", mPath)
+		return "", errors.New(errorMsg)
```
Comment (apelisse) [inline pkg/kinflate/commands/util.go]: You can use `fmt.Errorf` instead? (and everywhere else)

File: pkg/kinflate/commands/util.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	}
+	if f.IsDir() {
+		mPath = path.Join(mPath, constants.KubeManifestFileName)
+		_, err = fsys.Stat(mPath)
+		if err != nil {
```
Comment (apelisse) [inline pkg/kinflate/commands/util.go]: what if `mpath` is still a dir at this point? :D

File: pkg/kinflate/commands/util.go (hunk 1 above)
Comment (Liujingfang1) [reply pkg/kinflate/commands/util.go]: I'm OK with using a struct. How about 
```
type ManifestFile struct {
  mpath string
  fsys    fs.FileSystem
}

func New(path string, fsys fs.FileSystem) ManifestFile  //replace current validateManifestFile
func (ManifestFile) Read() (*Manifest, error)
func (ManifestFile) Write(m *Manifest) error
```

File: pkg/kinflate/commands/util.go (hunk 2 above)
Comment (Liujingfang1) [reply pkg/kinflate/commands/util.go]: Yes, that's true. The reason of keeping it this way is for future work on backlog item. We want to use the internal error package that Sean and I created previously. 

Actually, I can change this error to use manifest error.

File: pkg/kinflate/commands/util.go (hunk 3 above)
Comment (Liujingfang1) [reply pkg/kinflate/commands/util.go]: By the join statement with `constants.KubeManifestFileName`, mPath should be a file path.

File: pkg/kinflate/commands/util.go (hunk 1 above)
Comment (Liujingfang1) [reply pkg/kinflate/commands/util.go]: Actually, this struct and member functions should be private.

File: pkg/kinflate/commands/util.go (hunk 3 above)
Comment (apelisse) [reply pkg/kinflate/commands/util.go]: Well, unless you have a dir named `Kube-manifest.yaml`. Though unlikely, I agree ;-)

**Comment by Liujingfang1** [comment]
Add `manifestFile` struct and functions
```
newManifestfile()
validate()
read()
write()
```

**Pull request merged**